src/Symfony/Component/Validator/Constraints/NoSuspiciousCharactersValidator.php

```diff
@@ -94,24 +94,12 @@ public function validate(mixed $value, Constraint $constraint): void
 
         $checker->setChecks($checks);
 
-        if (!$checker->isSuspicious($value)) {
-            return;
-        }
-
-        foreach (self::CHECK_ERROR as $check => $error) {
-            if (!($checks & $check)) {
-                continue;
-            }
-
-            $checker->setChecks($check);
-
-            if (!$checker->isSuspicious($value)) {
-                continue;
-            }
+        if ($checker->isSuspicious($value, $error)) {
+            ['messageProperty' => $messageProperty, 'code' => $code] = self::CHECK_ERROR[$error];
```

Comment:
Should we check here first that the array has the key `$error`?

Reply:
This should not be necessary, but I just noticed `$error` is a bitmask so this PR needs more work.